src/main/java/fr/inria/jtravis/helpers/RepositoryHelper.java

```diff
@@ -18,16 +21,40 @@ public RepositoryHelper(JTravis jTravis) {
     }
 
     public Optional<Repository> fromSlug(String slug) {
+        Optional<Repository> repo = null;
+
         Optional<String> optionalS = this.getEncodedSlug(slug);
         if (optionalS.isPresent()) {
-            return getEntityFromUri(Repository.class, Arrays.asList(
+            repo = getEntityFromUri(Repository.class, Arrays.asList(
                     TravisConstants.REPO_ENDPOINT,
                     optionalS.get()), null);
+            if (!repo.isPresent()) {
+                String newSlug = searchCurrentRepoNameWithGHAPI(slug);
+                if (newSlug != null && !newSlug.equals(slug)) {
+                    repo = this.fromSlug(newSlug);
+                }
+            }
+        }
+        if (repo != null && repo.isPresent()) {
+            return repo;
         } else {
             return Optional.empty();
         }
     }
 
+    protected String searchCurrentRepoNameWithGHAPI(String slug) {
+        JTravis jTravis = JTravis.builder().build();
+        GitHub gitHub;
+        try {
+            gitHub = jTravis.getGithub();
+            GHRepository repository = gitHub.getRepository(slug);
```

Comment:
We should check that slug and repository.getFullName are different and inform the user through a log